File: langgraph/data/equiv/145.py
def sort_numbers_by_digit_sum(numbers):
    def calculate_weight(number):
        sign = -1 if number < 0 else 1
        absolute_value = abs(number)
        digits = [int(digit) for digit in str(absolute_value)]
        digits[0] *= sign
        return sum(digits)
    
    return sorted(numbers, key=calculate_weight)

# method name: order_by_points -> sort_numbers_by_digit_sum
# parameter name: nums -> numbers
# inner function name: digits_sum -> calculate_weight
# changed variable names: n -> number, neg -> sign, n -> digits
# used abs() instead of manually multiplying by -1
# simplified the condition for sign assignment

File: langgraph/result/qwen/32b-py2js/145.js
function orderByPoints(nums) {
    function digitsSum(n) {
        let neg = 1;
        if (n < 0) {
            n = -1 * n;
            neg = -1;
        }
        const digits = Array.from(String(n), Number);
        digits[0] = digits[0] * neg;
        return digits.reduce((sum, digit) => sum + digit, 0);
    }
    return nums.sort((a, b) => digitsSum(a) - digitsSum(b));
}